### PR TITLE
Benchmark

### PR DESCRIPTION
**Pull request by Karajan421**
This includes the small optimization of add_file function and lay the groundwork for embedding models benchmark

### CODE REVIEW

File: app/memory/chroma_memory.py [hunk 1]
```diff
@@ -2,20 +2,36 @@
...
 
+instruct_embeddings=["hkunlp/instructor-xl","hkunlp/instructor-large"]
+sentence_transformers_embeddings=["all-MiniLM-L6-v2","sentence-t5-xxl"]
 
 class Chroma(BaseMemory):
     vector_store: Optional[Type[vectorstores.Chroma]]
     collection_name: Optional[str]
 
+    embeddings_map = {
+        **{name: HuggingFaceInstructEmbeddings for name in instruct_embeddings},
```
Comment (CryptoRUSHGav) [inline app/memory/chroma_memory.py:19]: nice usage of the dictionary unpacking operator!

File: misc/avoid_warning_linux.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            else:
+                print(line, end='')
+
+
+# Get the username of the current user
+username = getpass.getuser()
+# Get the user's Python version
+python_version = f"{sys.version_info.major}.{sys.version_info.minor}"
+# Construct the path to the script based on the username and Python version
+script_file = f"/home/{username}/anaconda3/envs/brainchulo/lib/python{python_version}/site-packages/chromadb/api/models/Collection.py"
```
Comment (CryptoRUSHGav) [inline misc/avoid_warning_linux.py:21]: we should not assume everyone uses conda.

Comment (CryptoRUSHGav) [reply misc/avoid_warning_linux.py:21]: let's ponder whether this file is really needed.

**Review by CryptoRUSHGav — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -92,33 +90,30 @@ def bot(history):
...
+
+    btn_submit.click(add_text, [chatbot, txt], [chatbot, txt]).then(
+        bot, chatbot, chatbot
+    )
+    txt.submit(add_text, [chatbot, txt], [chatbot, txt]).then(
+        bot, chatbot, chatbot
+    )
+    btn.upload(add_file, [chatbot, btn], [chatbot])
+
+app.launch(server_port=7865, server_name="0.0.0.0", debug=True)
```
[the remark this answers is not in the document]
Comment (Karajan421) [reply app/main.py:119]: Got it, I'll go back on this! (and I'll keep an eye to see if we can find another solution to this docker issue)

File: misc/avoid_warning_linux.py (hunk 2 above)
Comment (Karajan421) [reply misc/avoid_warning_linux.py:21]: Yeah I'll remove it from the PR, the loggerhandler solution seems much better

**Review by CryptoRUSHGav — APPROVED** [review]
perfection!

**Pull request merged**